<launch>
    <rosparam command="load" file="$(find rgbd_object_detection)/config/realsense.yaml" />

    <node name="rviz" pkg="rviz" type="rviz"  args="-d $(find rgbd_object_detection)/rviz/test_frames.rviz"/>
    
    <node name="run_object_detection" pkg="rgbd_object_detection" type="run_object_detection" output="screen"/>

    <!-- Learned: Euler angle is not fixed angle -->
    <!-- Learned: Translation first then rotation Rcl_ * p_l + tcl_ -->

    <!-- ############################ Testing ################################### -->

    <!-- Initial guess -> Euler angles of multiple axis rotations (degrees) x:0 y:-110 z:90 -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="camera2tmp" args="0 0 0 0.579228 -0.579228 0.4055798 0.4055798 camera_color_optical_frame tmp 100"/> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="tmp2fake_lidar" args="0.3 0 -1.3 0 0 0 tmp fake_lidar 100"/> -->
   
    <!-- By doing rosrun tf tf_echo fake_lidar camera_link -->
    <!-- Camera to lidar is 0.724, 0.014, -1.120 RPY: 0.001, -0.348, -0.002-->
    <!-- <node pkg="tf" type="static_transform_publisher" name="lidar2camera" args="-0.299 -0.015 1.300 0.002 0.348 0.000 fake_lidar camera_link 100"/> -->
    
    <!-- Test velodyne to camera -->
    <!-- Unable to solve too old message problem, tried use_sim_time and clock tag when play bag-->
    <!-- <node pkg="tf" type="static_transform_publisher" name="velodyne2camera" args="-0.299 -0.015 1.300 0.002 0.348 0.000 velodyne camera_link 100"/> -->

    <!-- ############################ Final ################################### -->

    <node pkg="tf" type="static_transform_publisher" name="camera2color_optical" args="-0.001 0.015 -0.000 -1.572 -0.002 -1.571 camera_link camera_color_optical_frame 100"/>
    <node pkg="tf" type="static_transform_publisher" name="camera2depth_optical" args="0.000 0.000 0.000 -1.571 -0.000 -1.571 camera_link camera_depth_optical_frame 100"/>

    <node pkg="tf" type="static_transform_publisher" name="camera2tmp" args="0 0 0 0.579228 -0.579228 0.4055798 0.4055798 camera_color_optical_frame tmp 100"/>
    <node pkg="tf" type="static_transform_publisher" name="tmp2velodyne" args="0.15 0 -0.95 0 0 0 tmp velodyne 100"/>
    



</launch>